### ISSUE TITLE
hlint-action error: libtinfo.so.5: cannot open shared object file

### ISSUE THREAD

**Issue by pbrisbin** [haskell/actions#128]
GitHub has updated `ubuntu-latest`,

![](https://files.pbrisbin.com/screenshots/screenshot.471187.png)

We're bumping into issues here and there and one is the `hlint-run` Action failing,

```
/opt/hostedtoolcache/hlint/3.3.6/x64/hlint: error while loading shared libraries: libtinfo.so.5: cannot open shared object file: No such file or directory
hlint completed with status code 127
```

I've worked around by adding the following step first,

```yaml
- run: sudo apt-get install -y libncurses5
```

I'm not sure if `hlint-setup` should include this?

Annoying because it added ~5s to a Job that's only ~4s in total normally.

**Comment by pbrisbin** [comment]
FYI I'm now finding that `ubuntu-latest` == `ubuntu-22.04` may not be true yet. We're seeing it, but it's inconsistent and not what the official docs say. So I guess consider this a bug report for anyone using `ubuntu-22.04` explicitly, but I think that'll be `ubuntu-latest` soon.

**Comment by andreasabel** [comment]
@pbrisbin : Consider reporting this upstream at https://github.com/actions/runner-images/issues if you think that `libncurses5` should come by default with `ubuntu-22.04`.  But maybe something in this action can be bumped to [libncurses 6](https://launchpad.net/ubuntu/jammy/+package/libncurses-dev).

**Comment by andreasabel** [comment]
> So one thought could be to move the action's default `version` up to 3.5 now that `ubuntu-latest` is 22.04. This way defaults-with-defaults works again.

Sounds reasonable.  

With the caveat that `ubuntu-latest` is sometimes still `ubuntu-20.04`.  According to this issue, the migration will only be completed in March 2023:
- https://github.com/actions/runner-images/issues/6399

**Comment by pbrisbin** [comment]
Ah thanks for linking that! We've been seeing the intermittent behavior and I'm happy to see an official source that it's due to an ongoing rollout.

**Comment by andreasabel** [comment]
@rwe: Are you maintaining the `hlint` actions?

**Comment by rwe** [comment]
@andreasabel I haven't been touching them since the fork here—they were subtreed so I didn't see it actually occur. And despite being the author I don't have commit access here.

But I might have some time to take a look and review today if there's anything I can contribute.

**Comment by andreasabel** [comment]
@rwe : I think @hazelweakly can give you commit rights.  I guess she would be happy to get some maintenance help.

**Comment by andreasabel** [comment]
@rwe: I lack the rights to give you rights, but I can merge and publish a PR if you provide one.

**Comment by andreasabel** [comment]
Released fix as v2.0.4: bump hlint default version to 3.5.
